### DISCUSSION TITLE
Allow querying/filtering DOI by status (findable, registered, or draft) via the API

### DISCUSSION THREAD

**Discussion by eltiffster** [datacite/datacite-suggestions#110] [General Suggestions]
### What is the problem that your suggestion solves?

While you can filter DOIs by status in the Fabrica interface, it seems this can't be replicated through the API. There is nothing in the [API documentation](https://support.datacite.org/docs/api-queries#query-examples) that suggests this is possible, and using the query parameter "query=state:registered" or "query=state:draft" doesn't return any results. (If there's already an option for this that I've missed, please let me know.)

Without the above functionality, someone would need to fetch _all DOIs_ and then filter them all by the nested state attribute. This is less efficient than simply querying by status in the first place, which would return only matching DOIs. Adding a "state" filter to the API would be especially beneficial for repositories that contain a large number of DOIs.

I believe adding a "state" filter would meet the following potential use cases, among others:

- Selecting all registered or draft DOIs in order to systematically change them to findable (or registered)
- Deleting multiple draft DOIs (e.g. that were erroneously created by an automated third-party process/script)
- Any other bulk action that should only affect registered or draft DOIs

### What solution might meet your needs?

Changing the API to accept a "status" filter parameter would be great. That way, I could simply append "state:registered" to any query and get the expected results back.

### Your name

Tiffany Chan

### Your organization

University of Victoria Libraries

### What alternatives have you tried or considered?

Grabbing all DOIs, 1000 at a time, then filtering them by the "state" attribute. Since the API can only return a maximum of 1000 DOIs at a time and we have over 13,000 DOIs, this takes a while.

By contrast, we only have 102 registered DOIs. So it would be more efficient to make one API request for the 102 registered DOIs instead of 14 API requests.

### Is there anything else you would like to share?

Thanks for considering!

### What group(s) would benefit from your suggestion?

- [X] DataCite members registering DOIs and metadata and administering accounts
- [X] Developers building integrations with DataCite services
- [ ] Harvesters who index DataCite DOIs and metadata
- [X] Individuals or institutions analyzing DataCite DOIs and metadata
- [ ] DataCite staff
- [ ] Other

**Comment by KellyStathis** [comment]
Hi @eltiffster, there is existing functionality for this! The dois endpoint has a `state` filter, which is documented in our API Reference: https://support.datacite.org/reference/get_dois. Note that retrieving registered state DOIs and draft records requires authentication.

If you need any assistance, feel free to reach out to us at support@datacite.org.

**Reply by eltiffster** [reply]
Ah, I was using the wrong syntax. Thank you for the quick reply and for pointing me in the right direction!
